### PR TITLE
Adding  `application.giantswarm.io/team` label

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/24111

Also moving to app-build-suite

### CODE REVIEW

File: .abs/main.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+replace-chart-version-with-git: true
+generate-metadata: true
+chart-dir: ./helm/vertical-pod-autoscaler-app
+destination: ./build
+ct-config: ./.circleci/ct-config.yaml
+catalog-base-url: https://giantswarm.github.io/giantswarm-catalog/
```
Comment (QuentinBisson) [inline .abs/main.yaml:6]: Missing EOL :)

File: .kube-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+checks:
+  exclude:
+  - "no-read-only-root-fs"
+  - "run-as-non-root"
+  - "unset-cpu-requirements"
+  - "unset-memory-requirements"
+  - "no-anti-affinity"
+  - "latest-tag"
```
Comment (QuentinBisson) [inline .kube-linter.yaml:8]: I'm not sure I like that but that's how the app is already 👍🏻

**Review by QuentinBisson — APPROVED** [review]
Apart from the EOL, LGTM if tested :)

File: .kube-linter.yaml (hunk 2 above)
Comment (marieroque) [reply .kube-linter.yaml:8]: Yes I asked myself the same question.
Let's begin with this config and we'll tune later...

**Review by hervenicol — APPROVED** [review]
LGTM

**Comment by hervenicol** [comment]
Note: we didn't manage to match team label with SLO metrics (because of strange pod names), so we ended up hardcoding team in the SLO metrics :disappointed: https://github.com/giantswarm/prometheus-rules/pull/517

**Comment by QuentinBisson** [comment]
@hervenicol do you remember the strange pod name?

**Comment by hervenicol** [comment]
> @hervenicol do you remember the strange pod name?

If I remember correctly, the name for app-updater pods was "writer" or something like that.
I did some quick `kubectl` on a cluster but did not find evidence of what we had seen on the metrics with @marieroque.
I should dig back into the metrics for that.

**Review by TheoBrigitte — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Adding `application.giantswarm.io/team` label
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply CHANGELOG.md:12]: to kubernetes resources ;)

**Comment by marieroque** [comment]
> uld dig back into the metrics for that

The `app` label in `up` metrics was related to the service name (vertical-pod-autoscaler-app-webhook) and not the deployment name as expected.
As the `application.giantswarm.io/team` label is only propagated to `kube_deployment_labels` metrics, we were not able to retrieve the team label.
Hence we have set statically the team in the recording rules :-(

**Pull request merged**